### PR TITLE
Fix part of #2294: Migrating to TextInputLayout for ProfileRenameActivity

### PR DESCRIPTION
**Pull request by FareesHussain**
## Explanation

Fix part of #2294 
Fix part of #962 
Migrating for `ProfileRenameActivity`

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

## ScreenShots
Portrait | Landscape
--- | ---
image | image
image | image
image | image

## Video

profileRenameActivity

## Tests
**Espresso**
image

**Robolectric**

image

### CODE REVIEW

**Comment by anandwana001** [comment]
Please follow all the points in issues How to submit PR for this issue.

File: app/src/main/res/layout-land/profile_rename_activity.xml [hunk 1]
```diff
@@ -51,22 +50,33 @@
...
+          app:layout_constraintTop_toTopOf="parent">
+
+          <com.google.android.material.textfield.TextInputEditText
+            android:id="@+id/profile_rename_input_edit_text"
+            style="@style/TextInputEditText"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="@string/profile_rename_label"
+            android:imeOptions="actionDone"
+            android:maxLength="36"
```
Comment (anandwana001) [inline app/src/main/res/layout-land/profile_rename_activity.xml:77]: how are we defining 36 here?

File: app/src/main/res/layout/profile_rename_activity.xml [hunk 2]
```diff
@@ -55,18 +54,30 @@
...
+          app:layout_constraintTop_toTopOf="parent">
+
+          <com.google.android.material.textfield.TextInputEditText
+            android:id="@+id/profile_rename_input_edit_text"
+            style="@style/TextInputEditText"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:maxLength="36"
+            android:singleLine="true"
+            android:hint="@string/profile_rename_label"
```
Comment (anandwana001) [inline app/src/main/res/layout/profile_rename_activity.xml]: reformat and re-arrange

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks, @FareesHussain Doing a quick review,
nit changes, and please add test results screenshot

File: app/src/main/res/layout-land/profile_rename_activity.xml (hunk 1 above)
Comment (FareesHussain) [reply app/src/main/res/layout-land/profile_rename_activity.xml:77]: @anandwana001 Actually this was missing.
for AddProfile the maxLength is 36 for the name so it must be present in Rename too.

File: app/src/main/res/layout/profile_rename_activity.xml (hunk 2 above)
Comment (FareesHussain) [reply app/src/main/res/layout/profile_rename_activity.xml]: Done

File: app/src/main/res/layout-land/profile_rename_activity.xml (hunk 1 above)
Comment (anandwana001) [reply app/src/main/res/layout-land/profile_rename_activity.xml:77]: nice catch.

**Comment by anandwana001** [comment]
Do we have something in TextInputLayout so that user can enter only letters not the number, just like in pin we can enter only numbers not text. We can apply the same here for text only. 
@rt4914 WDYT?

**Comment by FareesHussain** [comment]
> Do we have something in TextInputLayout so that user can enter only letters not the number, just like in pin we can enter only numbers not text. We can apply the same here for text only.
> @rt4914 WDYT?

Can we try [this](https://stackoverflow.com/a/20868656/12088301)

**Comment by anandwana001** [comment]
As we already had a check for this that if the user will enter the number the error will be shown. 
@FareesHussain Does it work with `TextInputEditText`?

**Comment by FareesHussain** [comment]
> As we already had a check for this that if the user will enter the number the error will be shown.
> @FareesHussain Does it work with `TextInputEditText`?

Yes, in the gif you can see the error message

**Comment by FareesHussain** [comment]
@anandwana001 do I have to add only letters for textInputLayout in this pr?
I think it might be better to create a separate issue for that

**Comment by anandwana001** [comment]
> @anandwana001 do I have to add only letters for textInputLayout in this pr?
> I think it might be better to create a separate issue for that

Yes, let's create a separate issue, will discuss there if we need it or not.

**Comment by anandwana001** [comment]
Explanation and PR title is different

**Comment by FareesHussain** [comment]
> Explanation and PR title is different

Updated

**Review by anandwana001 — APPROVED** [review]
Thanks @FareesHussain LGTM
Is it all migration thing going good with Tablet too?

**Comment by FareesHussain** [comment]
> Thanks @FareesHussain LGTM
> Is it all migration thing going good with Tablet too?

Yes, I'll update the description for Tablet Screenshots too.

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**